substrate/frame/treasury/src/lib.rs

```diff
@@ -289,6 +293,9 @@ pub mod pallet {
 		/// Helper type for benchmarks.
 		#[cfg(feature = "runtime-benchmarks")]
 		type BenchmarkHelper: ArgumentsFactory<Self::AssetKind, Self::Beneficiary>;
+
+		/// Provider for the block number.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
may be a comment that this is the system pallet in general case

Reply:
done